edgexfoundry/device-rfid-llrp-go
# refactor(snap): edgex-snap-hooks related upgrade

This PR upgrades snap hooks to [2.4.1 ](https://github.com/canonical/edgex-snap-hooks/releases/tag/v2.4.1)with new function and refactor:
- Combine install hook and configure hook
- Use library function for copying directory
- Remove export bash command to process valid .env files
- Use log package for logging
- Use library function for autostart processing

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-rfid-llrp-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?) [device-rfid-llrp test suite](https://github.com/canonical/edgex-snap-testing/tree/main/test/suites/device-rfid-llrp) passed with `FULL_CONFIG_TEST=true`
  <link to docs PR>

## Testing Instructions

1. Install and validate that all config files have been copied in right place:
```
tree /var/snap/edgex-device-rfid-llrp/current/config
.
└── device-rfid-llrp
    └── res
        ├── configuration.toml
        ├── devices
        ├── profiles
        │   ├── llrp.device.profile.yaml
        │   └── llrp.impinj.profile.yaml
        └── provision_watchers
            ├── impinj.provision.watcher.json
            └── llrp.provision.watcher.json

5 directories, 5 files
```
2. Setup and start:
```
sudo snap set edgex-device-rfid-llrp app-options=true
sudo snap set edgex-device-rfid-llrp config.writable-loglevel=DEBUG
sudo snap set edgex-device-rfid-llrp config.service-startupmsg="#test start up message#"
sudo snap start edgex-device-rfid-llrp
```

3. Check the .env file is valid now without `export` command:
```
cat /var/snap/edgex-device-rfid-llrp/current/config/device-rfid-llrp/res/device-rfid-llrp.env 

SERVICE_STARTUPMSG="#test start up message#"
WRITABLE_LOGLEVEL="DEBUG"
```
4. Validate env vars in `.env` have been exported successfully:
```
sudo snap logs -f edgex-device-rfid-llrp

msg="#test start up message#"
level=DEBUG msg="xxx"
```
5. Validate autostart via snap option works:
```
$ sudo snap set edgex-device-rfid-llrp autostart=true
$ snap services edgex-device-rfid-llrp
Service                                  Startup  Current  Notes
edgex-device-rfid-llrp.device-rfid-llrp  enabled  active   -

$ sudo snap set edgex-device-rfid-llrp autostart=false
$ snap services edgex-device-rfid-llrp
Service                                  Startup   Current   Notes
edgex-device-rfid-llrp.device-rfid-llrp  disabled  inactive  -
```

## New Dependency Instructions (If applicable)
